Minio/DataModel/BucketOperationsArgs.cs

```diff
@@ -157,24 +181,41 @@ public GetObjectListArgs WithContinuationToken(string token)
             return this;
         }
 
+        public GetObjectListArgs WithListObjectsV1(bool useV1)
+        {
+            this.UseV2 = !useV1;
+            return this;
+        }
+
         internal override RestRequest BuildRequest(RestRequest request)
         {
             if (this.Versions)
             {
                 request.AddQueryParameter("versions", "");
             }
-            else
+            else if(this.UseV2)
             {
                 request.AddQueryParameter("list-type", "2");
             }
             request.AddQueryParameter("delimiter",this.Delimiter);
-            request.AddQueryParameter("prefix",this.Prefix);
+            if (!string.IsNullOrWhiteSpace(this.Prefix))
+            {
+                request.AddQueryParameter("prefix",this.Prefix);
+            }
             request.AddQueryParameter("max-keys", "1000");
             request.AddQueryParameter("encoding-type","url");
-            if (!string.IsNullOrWhiteSpace(this.KeyMarker))
+            if (!string.IsNullOrWhiteSpace(this.KeyMarker) && this.UseV2)
```

Comment:
useV2 is not important here, versions is important here

Reply:
Changed. PTAL.